src/context.ts

```diff
@@ -47,6 +48,43 @@ export interface DocumentTypeOption {
   };
 }
 
+export interface ServiceFetchConfiguration {
```

Comment:
hrm, I'm not sure this belongs in the form-builder repository - does something break if you ignore `serviceFetchConfiguration` in the `VariableDefinition` type?

Perhaps it makes more sense to pass `getInfernoLogicContext` via the builder context rather than providing the inputs and creating this context? That way we can build that context in our backend JS and pass it to the builder, and the open-forms specific stuff is more decoupled from the formio domain for the builder? Ultimately you sort of initialize the builder with "here, this is the infernologic type checking context, use that for your type checks and please validate the things in the builder" without the builder itself needing to understand what `infer` needs (other than the function signature).